src/components/purchase/Erc20TokenStep.vue

```diff
@@ -0,0 +1,97 @@
+<template>
+  <div class="erc20-token tile is-hcentered">
+    <div
+      class="create-token tile is-8"
+      v-if="needsToken">
+      <div class="indicator tile is-2">
+        <ProcessButton
+          :processing="isProcessing"
+          :onClickCallback="onClickCallback"/>
+      </div>
+      <div class="label tile">
+        {{ labelText }}
+      </div>
+    </div>
+    <div
+      class="market-token-balance tile is-8"
+      v-else>
+      <div class="indicator tile is-2"></div>
+      <div class="label tile">
+        {{ marketTokenBalance }}
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop, Watch } from 'vue-property-decorator'
+import { NoCache } from 'vue-class-decorator'
+import { MutationPayload, Store } from 'vuex'
+import { VuexModule, getModule } from 'vuex-module-decorators'
+
+import AppModule from '../../vuexModules/AppModule'
+import Web3Module from '../../vuexModules/Web3Module'
+import PurchaseModule from '../../vuexModules/PurchaseModule'
+
+import ProcessButton from '@/components/ui/ProcessButton.vue'
+
+import { PurchaseStep } from '../../models/PurchaseStep'
+import FfaListing from '../../models/FfaListing'
+
+import PurchaseProcessModule from '../../functionModules/components/PurchaseProcessModule'
+import EtherTokenContractModule from '../../functionModules/protocol/EtherTokenContractModule'
+
+import { Labels } from '../../util/Constants'
+
+@Component({
+  components: {
+    ProcessButton,
+  },
+})
+export default class Erc20TokenStep extends Vue {
+
+  @NoCache
+  public get needsToken(): boolean {
+    const purchaseModule = getModule(PurchaseModule, this.$store)
```

Comment:
Possible convention change- can call modules as data a single time and use `this.moduleName` throughout the component instead. Removes the need for ad-hoc `getModule` calls.